src/lib/Libnet/get_hostaddr.c

```diff
@@ -146,10 +147,23 @@ compare_short_hostname(char *shost, char *lhost)
 {
 	size_t   len;
 	char    *pdot;
+	int	is_shost_ip;
+	int	is_lhost_ip;
+	struct	sockaddr_in check_ip;
 
 	if ((shost == NULL) || (lhost == NULL))
 		return 1;
 
+	/* check if hostnames given are in IPV4 dotted-decimal form: ddd.ddd.ddd.ddd */
+	is_shost_ip = inet_pton(AF_INET, shost, &(check_ip.sin_addr));
+	is_lhost_ip = inet_pton(AF_INET, lhost, &(check_ip.sin_addr));
+	if ((is_shost_ip > 0) || (is_lhost_ip > 0)) {
+		if (strcasecmp(shost, lhost) == 0)
```

Comment:
We know at least one of the strings is the dotted decimal representation, and we know the maximum length of the string is ((3 * 4) + 3) = 15 characters. We should be able to do a strncmp() of length 15. Not a huge performance improvement, but the right tool for the job, IMHO. Otherwise, it looks good.

Reply:
I like your suggestion. Ok, I'll incorporate.